### ISSUE TITLE
Create modules for subspecs

### ISSUE THREAD

**Issue by mozeryansky** [CocoaPods/CocoaPods#8413]
I'd like cocoapods to generate a modulemap when I use subspecs. I thought this was how it was done to match the concept of a submodule from modulemap. Is this possible or considerable?

For example (pseudocode):

Podspec:
```
Pod:: do |s|
    s.name = 'MyPod'
    s.modules = true  # <= potential configuration?
    s.subspec 'SubPodA' do |ss|
        ss.source_files = '...'
    end
    s.subspec 'SubPodB' do |ss|
        ss.source_files = '...'
    end
end
```

Then when I use `pod 'MyPod'` I'd like the resulting modules map to be:
```
framework module MyPod {
  module SubPodA { export ... }
  module SubPodB { export ... }
}
```

Then I can import in my project `MyPod.SubPodA`.

I imagine if someone uses `pod 'MyPod/SubPodA'` the the resulting modulemap would be normal:
```
framework module SubPodA {
    ...
}
```

**Comment by dnkoutso** [comment]
I dont think this is supported right now as you point out. Best workaround is to provide your own module map for now.
